spec/bing_translator_spec.rb

```diff
@@ -62,12 +65,54 @@ def load_file(filename)
       expect(result).to eq message_en
     end
 
+    context 'when the authentication server is offline' do
+      it 'throws a reasonable error when a 500 error is encountered' do
+        stub_request(:any, COGNITIVE_ACCESS_TOKEN_URI).
```

Comment:
I made a change on master that reuses tokens across RSpec examples in order to preview 429s on the authentication endpoint; which interferes with this because the token endpoint isn't rehit and the stub goes unused.

Based on the change I already authored, `let`ting a different token would bust the cache. Perhaps a more proper fix would involve replacing my change with a stub on the token endpoint, but it's rather involved. Let me know which way you're more comfortable with.

Reply:
I merged in your change and then updated my tests.  For the tests that need to hit the authentication, I create a new instance of BingTranslator without the cached access token.

There is only one test in my PR that actually hits the authentication server--the rest are all guarded with mocks--so hopefully that shouldn't trigger any 429 errors?

This was done in 51e2847

Please let me know what you think.

Reply:
Looks good, the long-running 429 issue in our CI is something that isn't too pretty to handle in a clean testing way, so I don't expect fixes for it to feel great either. Thanks for handling this moving target